APIScrape/scrape.php

```diff
@@ -0,0 +1,177 @@
+ */
+function getCatImageURLs(array $catBreeds):array
+{
+    $imageSourceArray = [];
+    $breedCount = count($catBreeds);
+    $i = 1;
+    echo "Fetching cat image URLs...\n";
+    foreach ($catBreeds as $id => $name) {
+        $catImageApiUrl = 'https://api.thecatapi.com/v1/images/search?breed_ids=' . $id . '&limit=21';
+        $curl = curl_init();
+        curl_setopt_array($curl, array(
+            CURLOPT_URL => $catImageApiUrl,
+            CURLOPT_CUSTOMREQUEST => "GET",
+            CURLOPT_RETURNTRANSFER => true,
+            CURLOPT_HTTPHEADER => [ API_KEY ]
+        ));
+        $imageApiResponse = curl_exec($curl);
+        if($imageApiResponse === false) {
+            echo 'Curl error: ' . curl_error($curl);
+        } else {
+            // Little bash technique to clean up console reporting
+            // Use bash escape \033[2K to clear the line before writing the message
+            // \r carriage returns to the beginning of the line before writing the next
+            // The console lines write in-place - much neater
+            echo "\033[2KReceived cat image URLs for breed '$name' successfully ($i of $breedCount).\r";
+        }
+        $i++;
+
+        curl_close($curl);
+
+        $responseArray = json_decode($imageApiResponse, true);
+
+        $breedImages = [];
+        foreach($responseArray as $item) {
+            array_push($breedImages, $item["url"]);
+        }
+        $imageSourceArray[$name] = $breedImages;
+    }
+    return $imageSourceArray;
+}
+
+/**
+ * Puts the cat image URLs into the database.
+ * 
+ * @param PDO $db the database to use
+ * @param array The list of cat breeds
+ * @param array The list of cat image URLs
+ * @return void
+ */
+function fillCatImages(PDO $db, array $catBreedArray, array $catImageSourceArray)
+{
+    $catBreedIndexedArray = [];
+    foreach($catBreedArray as $breed) {
+        $catBreedIndexedArray[] = $breed;
+    }
+    $sqlArray = [];
+    for($breedIndex = 0; $breedIndex < count($catImageSourceArray); $breedIndex++) {
+        foreach($catImageSourceArray[$catBreedIndexedArray[$breedIndex]] as $url) {
+            $breedID = $breedIndex + 1;
+            $sqlArray[$url] = $breedID;
+        }
+    }
+
+    foreach($sqlArray as $url=>$breedID) {
+        $sql = $db->prepare('INSERT INTO `img` (img_src, breed_id) VALUES (:url, :breedID);');
+        $sql->bindParam('url', $url, PDO::PARAM_STR);
+        $sql->bindParam('breedID', $breedID, PDO::PARAM_STR);
+        $sql->execute();
+    }
+    // Use bash escape \033[2K to clear the line before writing the message
+    // This console line overwrites what was previously there
+    echo "\033[2KAll cat image URLs added to database.\n";
+}
+
+// Do the business:
+createDatabase($dbConnection);
+$breeds = getCatBreeds();
+fillCatBreedToDB($dbConnection, $breeds);
+$catImageSourceArray = getCatImageURLs($breeds);
+fillCatImages($dbConnection, $breeds, $catImageSourceArray);
```

Comment:
Just putting this down for posterity:
Ultimately, I'm happy with this like this, because now we're not calling `getCatBreeds` twice. We discussed that I'd prefer a more procedural file because this is a one-off, manually run script. A lot of these functions are redundant (no potential for reuse) and the file could be shorter.
Good work! 👍